src/components/CCPDashboard/PatientInfoTable.tsx

```diff
@@ -155,9 +161,22 @@ export const PatientInfoTable = ({
     patients.find((x) => x.id === patientId)
   );
   const [anchorEl, setAnchorEl] = React.useState(null);
+  const [runNumber, setRunNumber] = React.useState<number | null>(
+    selectedPatient ? selectedPatient.runNumber : null
+  );
+  const [isNewRunNumber, setIsNewRunNumber] = React.useState<boolean>(false);
   const history = useHistory();
+  const { enqueueSnackbar } = useSnackbar();
 
-  const [editPatient] = useMutation(EDIT_PATIENT);
+  const [editPatient] = useMutation(EDIT_PATIENT, {
+    update(cache) {
+      const patientId = ((selectedPatient as unknown) as Patient).id;
+      const { patients } = cache.readQuery<any>({
+        query: GET_ALL_PATIENTS,
+      });
+      setSelectedPatient(patients.find((x) => x.id === patientId));
```

Comment:
This ensures that `selectedPatient` always contains a fresh state even when the modal is not closed and reopened. 

Without it, the `handleRunNumber` function will utilize the previous `runNumber` for comparison since `selectedPatient` is only set when the user opens a modal.

Reply:
just to double check, does this introduce any additional network requests by any chance? i checked my network tab and it seemed reasonable

Reply:
Nope - I don't believe so, according to this https://www.apollographql.com/docs/react/caching/cache-interaction/#readquery
![image](https://user-images.githubusercontent.com/41495076/98053324-4d45f700-1e06-11eb-8a47-9874fa503c45.png)